frontera/utils/misc.py

```diff
@@ -29,11 +32,11 @@ def load_object(path):
 
 
 def get_crc32(name):
-    return crc32(name) if type(name) is str else crc32(name.encode('utf-8', 'ignore'))
+    return crc32(to_bytes(name, 'utf-8', 'ignore'))
 
 
 def chunks(l, n):
-    for i in xrange(0, len(l), n):
+    for i in range(0, len(l), n):
```

Comment:
I tested a version based on `islice`, but it is slower. The grouper recipe shared by redapple is faster than the existing function but it introduces `None` values into the returned lists when they are not split evenly(example - `[1,2,3,4,5] -> [[1,2], [3,4],[5, None]])`. So we need to change the way the function is used in frontera(this will decrease the overall performance gain).